edit/edit.js

```diff
@@ -260,24 +260,25 @@ function initCodeMirror() {
   };
 
   // initialize global editor controls
-  function optionsHtmlFromArray(options) {
-    return options.map(opt => '<option>' + opt + '</option>').join('');
+  function optionsFromArray(parent, options) {
+    options.map(opt => $element({tag: 'option', textContent: opt}))
+      .forEach(opt => parent.appendChild(opt));
```

Comment:
This function works with a live element (it's in DOM), so multiple appendChild are slower than one, which is why I use DocumentFragment in such cases: it appends the elements as a flat list in one operation.

```js
const fragment = document.createDocumentFragment();
for (const opt of options) {
  fragment.appendChild($element({tag: 'option', textContent: opt}));
}
parent.appendChild(fragment);
```